client/channel.go

```diff
@@ -80,21 +81,52 @@ func (c *Client) CreateChannel(appID string, appType string, name string, descri
 		if _, err := c.KotsClient.CreateChannel(appID, name, description); err != nil {
 			return nil, err
 		}
-		return c.KotsClient.ListChannels(appID)
+		return c.KotsHTTPClient.ListChannels(appID, appSlug)
 	}
 
 	return nil, errors.New("unknown app type")
 }
 
-func (c *Client) GetChannelByName(appID string, appType string, name string, description string, createIfAbsent bool) (*types.Channel, error) {
+func (c *Client) GetOrCreateChannelByName(appID string, appType string, appSlug string, name string, description string, createIfAbsent bool) (*types.Channel, error) {
```

Comment:
This method is refactored from each individual Platform/Ship/KOTS client, because the logic is basically the same.